sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParameterNamesInPartialMethod.cs

```diff
@@ -48,29 +49,40 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     var methodSyntax = (MethodDeclarationSyntax)c.Node;
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodSyntax);
 
-                    if (methodSymbol?.PartialDefinitionPart == null)
+                    if (methodSymbol?.PartialDefinitionPart != null)
                     {
-                        return;
+                        VerifyParameters(c, methodSyntax, expectedParameters: methodSymbol.PartialDefinitionPart.Parameters);
                     }
-
-                    var implementationParameters = methodSyntax.ParameterList.Parameters;
-                    var definitionParameters = methodSymbol.PartialDefinitionPart.Parameters;
-
-                    for (var i = 0; i < implementationParameters.Count && i < definitionParameters.Length; i++)
+                    else if (methodSymbol?.OverriddenMethod != null)
                     {
-                        var implementationParameter = implementationParameters[i];
-
-                        var definitionParameter = definitionParameters[i];
-                        var implementationParameterName = implementationParameter.Identifier.ValueText;
-                        if (implementationParameterName != definitionParameter.Name)
-                        {
-                            c.ReportDiagnostic(Diagnostic.Create(rule,
-                                implementationParameter.Identifier.GetLocation(),
-                                implementationParameterName, definitionParameter.Name));
-                        }
+                        VerifyParameters(c, methodSyntax, expectedParameters: methodSymbol.OverriddenMethod.Parameters);
+                    }
+                    else
+                    {
+                        // do nothing
                     }
                 },
                 SyntaxKind.MethodDeclaration);
         }
+
+        private static void VerifyParameters(SyntaxNodeAnalysisContext context,
+            MethodDeclarationSyntax methodSyntax, IList<IParameterSymbol> expectedParameters)
+        {
+            var implementationParameters = methodSyntax.ParameterList.Parameters;

```

Comment:
Yes, otherwise the methods will not be matched as overrides... I replaced the loop with Zip.